tajo-client/src/main/java/org/apache/tajo/jdbc/TajoMemoryResultSet.java

```diff
@@ -18,46 +18,78 @@
 
 package org.apache.tajo.jdbc;
 
-import com.google.protobuf.ByteString;
+import io.netty.buffer.Unpooled;
 import org.apache.tajo.QueryId;
 import org.apache.tajo.catalog.Schema;
-import org.apache.tajo.storage.RowStoreUtil;
+import org.apache.tajo.catalog.SchemaUtil;
+import org.apache.tajo.exception.TajoInternalError;
+import org.apache.tajo.ipc.ClientProtos.SerializedResultSet;
 import org.apache.tajo.storage.Tuple;
+import org.apache.tajo.tuple.RowBlockReader;
+import org.apache.tajo.tuple.memory.HeapRowBlockReader;
+import org.apache.tajo.tuple.memory.HeapTuple;
+import org.apache.tajo.tuple.memory.MemoryBlock;
+import org.apache.tajo.tuple.memory.ResizableMemoryBlock;
+import org.apache.tajo.util.CompressionUtil;
 
 import java.io.IOException;
 import java.sql.SQLException;
-import java.util.List;
 import java.util.Map;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 public class TajoMemoryResultSet extends TajoResultSetBase {
-  private List<ByteString> serializedTuples;
-  private AtomicBoolean closed = new AtomicBoolean(false);
-  private RowStoreUtil.RowStoreDecoder decoder;
+  private MemoryBlock memory;
+  private RowBlockReader reader;
+  private volatile boolean closed;
 
-  public TajoMemoryResultSet(QueryId queryId, Schema schema, List<ByteString> serializedTuples, int maxRowNum,
+
+  public TajoMemoryResultSet(QueryId queryId, Schema schema, SerializedResultSet resultSet,
                              Map<String, String> clientSideSessionVars) {
     super(queryId, schema, clientSideSessionVars);
-    this.totalRow = maxRowNum;
-    this.serializedTuples = serializedTuples;
-    this.decoder = RowStoreUtil.createDecoder(schema);
+    if(resultSet != null && resultSet.getRows() > 0) {
+      this.totalRow = resultSet.getRows();
+
+      try {
+        // decompress if has a codec
```

Comment:
It should be ``decompress if it has a codec`` or ``decompress if a codec is specified``.